java-frontend/src/main/java/org/sonar/java/xml/maven/PomCheckContextImpl.java

```diff
@@ -67,15 +68,28 @@ public void reportIssue(PomCheck check, int line, String message, List<Location>
   }
 
   @VisibleForTesting
-  static AnalyzerMessage getSecondaryAnalyzerMessage(JavaCheck check, File file, Location location) {
-    XmlLocation startLocation = location.tree.startLocation();
-    XmlLocation endLocation = location.tree.endLocation();
-    TextSpan ts = new TextSpan(
-      startLocation.line(), offsetFromColumn(startLocation.column()),
-      endLocation.line(), offsetFromColumn(endLocation.column()));
+  AnalyzerMessage getSecondaryAnalyzerMessage(JavaCheck check, File file, Location location) {
```

Comment:
Unfortunately in order to compute the line size and get the correct text span I still need the access the `sonarComponents` object, which I don't want to see in the `Location` class. I'll let it that way. No need to rewrite the whole context for that simple case.